core-web/libs/block-editor/src/lib/extensions/ai-content-prompt/plugins/ai-content-prompt.plugin.ts

```diff
@@ -0,0 +1,242 @@
+import { Node } from 'prosemirror-model';
+import { EditorState, NodeSelection, Plugin, PluginKey, Transaction } from 'prosemirror-state';
+import { EditorView } from 'prosemirror-view';
+import { Subject } from 'rxjs';
+import tippy, { Instance, Props } from 'tippy.js';
+
+import { ComponentRef } from '@angular/core';
+
+import { takeUntil } from 'rxjs/operators';
+
+import { Editor, posToDOMRect } from '@tiptap/core';
+
+import { getNodePosition } from '../../bubble-menu/utils';
+import { AIContentPromptComponent } from '../ai-content-prompt.component';
+import { AI_CONTENT_PROMPT_PLUGIN_KEY } from '../ai-content-prompt.extension';
+import { TIPPY_OPTIONS } from '../utils';
+
+interface AIContentPromptProps {
+    pluginKey: PluginKey;
+    editor: Editor;
+    element: HTMLElement;
+    tippyOptions?: Partial<Props>;
+    component: ComponentRef<AIContentPromptComponent>;
+}
+
+interface PluginState {
+    open: boolean;
+    form: [];
+}
+
+export type AIContentPromptViewProps = AIContentPromptProps & {
+    view: EditorView;
+};
+
+export class AIContentPromptView {
+    public editor: Editor;
+
+    public node: Node;
+
+    public element: HTMLElement;
+
+    public view: EditorView;
+
+    public tippy: Instance | undefined;
+
+    public tippyOptions?: Partial<Props>;
+
+    public pluginKey: PluginKey;
+
+    public component?: ComponentRef<AIContentPromptComponent>;
+
+    private $destroy = new Subject<boolean>();
+
+    constructor(props: AIContentPromptViewProps) {
+        const { editor, element, view, tippyOptions = {}, pluginKey, component } = props;
+
+        this.editor = editor;
+        this.element = element;
+        this.view = view;
+
+        this.tippyOptions = tippyOptions;
+
+        this.element.remove();
+        this.element.style.visibility = 'visible';
```

Comment:
Are you sure that is ok to remove the element and then add styles?

Reply:
You are right, the sequence of removing the element and then setting its visibility doesn't have the desired effect in this context. I followed the pattern I observed in the bubble-link implementation. However, I now realize that my use-case might be different, and this sequence might not be necessary for the functionality I'm trying to achieve.

I appreciate your feedback and will ensure to be more attentive to such details in the future.